File: shopGunSdkDemo/src/androidTest/java/com/shopgun/android/sdk/demo/GeoHashTest.java
package com.shopgun.android.sdk.demo;

import android.location.Location;
import android.support.test.runner.AndroidJUnit4;

import com.fonfon.geohash.GeoHash;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;

/**
 * Simple tests for the geohash method we're using to convert the location
 */

@RunWith(AndroidJUnit4.class)
public class GeoHashTest {

    private final int PRECISION = 4;

    private Location location1;
    private Location location2;
    private Location location3;

    @Before
    public void setUp() {

        location1 = new Location("geohash");
        location1.setLatitude(55.679633);
        location1.setLongitude(12.577850);

        location2 = new Location("geohash");
        location2.setLatitude(38.354052);
        location2.setLongitude(-99.571113);

        location3 = new Location("geohash");
        location3.setLatitude(-31.386644);
        location3.setLongitude(131.341086);
    }

    @Test
    public void test_1() {
        GeoHash geoHash = GeoHash.fromLocation(location1, PRECISION);
        Assert.assertEquals("u3bu", geoHash.toString());
    }

    @Test
    public void test_2() {
        GeoHash geoHash = GeoHash.fromLocation(location2, PRECISION);
        Assert.assertEquals("9yc4", geoHash.toString());
    }

    @Test
    public void test_3() {
        GeoHash geoHash = GeoHash.fromLocation(location3, PRECISION);
        Assert.assertEquals("qfmm", geoHash.toString());
    }
}

File: tjekSdk/src/test/java/com/tjek/sdk/GeohashTest.kt
package com.tjek.sdk

import android.location.Location
import com.fonfon.geohash.GeoHash
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config

@RunWith(RobolectricTestRunner::class)
@Config(manifest = Config.NONE)
class GeohashTest {

    private val PRECISION = 4

    private var location1 = Location("geohash").apply {
        latitude = 55.679633
        longitude = 12.577850
    }
    private var location2 = Location("geohash").apply {
        latitude = 38.354052
        longitude = -99.571113
    }
    private var location3 = Location("geohash").apply {
        latitude = -31.386644
        longitude = 131.341086
    }

    @Test
    fun test_1() {
        val geoHash = GeoHash.fromLocation(location1, PRECISION)
        Assert.assertEquals("u3bu", geoHash.toString())
    }

    @Test
    fun test_2() {
        val geoHash = GeoHash.fromLocation(location2, PRECISION)
        Assert.assertEquals("9yc4", geoHash.toString())
    }

    @Test
    fun test_3() {
        val geoHash = GeoHash.fromLocation(location3, PRECISION)
        Assert.assertEquals("qfmm", geoHash.toString())
    }
}